src/timescale/push_github_events.service.ts

```diff
@@ -74,4 +75,38 @@ export class PushGithubEventsService {
 
     return rawResults as DbPushGitHubEventsHistogram[];
   }
+
+  async lastPushDatesForRepo(repo: string): Promise<{ push_date: Date; main_push_date: Date }> {
+    const startDate = GetPrevDateISOString(0);
+    const range = 30;
+
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder
+      .select(`*`)
+      .from("push_github_events", "push_github_events")
+      .where(`'${startDate}':: TIMESTAMP >= "push_github_events"."event_time"`)
+      .andWhere(`'${startDate}':: TIMESTAMP - INTERVAL '${range} days' <= "push_github_events"."event_time"`)
+      .andWhere(`LOWER("push_github_events"."repo_name") = '${repo}'`)
+      .orderBy("event_time", OrderDirectionEnum.DESC);
+
+    const results = await queryBuilder.getRawMany<DbPushGitHubEvents>();
+
+    if (results.length === 0) {
+      return { push_date: new Date(0), main_push_date: new Date(0) };
+    }
+
+    const firstPushDate = results[0].event_time;
+    const mainMasterRegex = new RegExp("main|master");
+    let firstMainPushDate = results.find((item) => item.push_ref && mainMasterRegex.test(item.push_ref))?.event_time;
+
+    if (!firstMainPushDate) {
+      firstMainPushDate = new Date(0);
+    }
```

Comment:
refactor (non-blocking): Consider using nullish coalescing instead, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing.

```suggestion
    const firstMainPushDate = results.find((item) => item.push_ref && mainMasterRegex.test(item.push_ref))?.event_time;
 ?? new Date(0);
```

Reply:
💯